discovery-client/src/main/java/io/micronaut/discovery/eureka/client/v2/AmazonInfo.java

```diff
@@ -249,17 +250,11 @@ public boolean equals(Object o) {
         if (this == o) {
             return true;
         }
-        if (!(o instanceof AmazonInfo)) {
+        if (!(o instanceof AmazonInfo that)) {
             return false;
         }
 
-        AmazonInfo that = (AmazonInfo) o;
-
-        if (metadata != null ? !metadata.equals(that.metadata) : that.metadata != null) {
-            return false;
-        }
-
-        return true;
+        return Objects.equals(metadata, that.metadata);
```

Comment:
It surprises me that this works...

I thought `that` was only in scope inside the `if` block...

Would 

```
        if (o instanceof AmazonInfo that) {
            return Objects.equals(metadata, that.metadata);
        }
        return false;
```

Be less surprising?